client/my-sites/checkout/src/lib/leave-checkout.ts

```diff
@@ -38,7 +38,11 @@ export const leaveCheckout = ( {
 	const launchpadURLRegex = /^\/setup\/[a-z][a-z\-_]*[a-z]\/launchpad\b/g;
 	const launchpadURLRegexMatch = redirectToParam?.toString().match( launchpadURLRegex );
 
-	if ( siteSlug && sendMessageToOpener( siteSlug, 'checkoutCancelled' ) ) {
+	if (
+		siteSlug &&
+		sendMessageToOpener( siteSlug, 'checkoutCancelled' ) &&
+		signupFlowName !== 'hosted-site-migration'
```

Comment:
Can we use the const names instead of the hardcoded ones they are on '@automattic/onboarding';
E.g. 
import { MIGRATION_FLOW } from '@automattic/onboarding';